lib/syskit/cli/log_runtime_archive_main.rb

```diff
@@ -0,0 +1,76 @@
+# frozen_string_literal: true
+
+# NOTE: this is NOT integrated in the Thor-based CLI to make it more independent
+# (i.e. not depending on actually having Syskit installed)
+
+require "pathname"
+require "thor"
+require "syskit/cli/log_runtime_archive"
+
+module Syskit
+    module CLI
+        # Command-line definition for the cli-archive-main syskit subcommand
+        class CLIArchiveMain < Thor
+            def self.exit_on_failure?
+                true
+            end
+
+            desc "watch", "watch a dataset root folder and call archiver"
+
+            option :period,
+                   type: :numeric, default: 600, desc: "polling period in seconds"
+            option :max_size,
+                   type: :numeric, default: 10_000, desc: "max log size in MB"
+            default_task def watch(root_dir, target_dir)
+                root_dir = validate_directory_exists(root_dir)
+                target_dir = validate_directory_exists(target_dir)
+                loop do
```

Comment:
You need to validate in #archive as well. Given the low frequency with which we'll call these methods, doing it only in #archive (removing it from watch) would be good enough.

You should test that the CLI fails if the directory does not exist.